contracts/Migrator.sol

```diff
@@ -0,0 +1,226 @@
+        if (endTime == 0 || isMigrationActive()) {
+            revert MigrationNotComplete();
+        }
+
+        emit Decommissioned();
+
+        uint256 ogvBalance = ogv.balanceOf(address(this));
+        if (ogvBalance > 0) {
+            // Burn all OGV
+            ogv.burn(ogvBalance);
+        }
+
+        uint256 ognBalance = ogn.balanceOf(address(this));
+        if (ognBalance > 0) {
+            // OGN doesn't allow burning of tokens. Has `onlyOwner`
+            // modifier on `burn` and `burnFrom` methods. Also,
+            // `transfer` has a address(0) check. So, this transfers
+            // everything to address(1). The `owner` multisig of
+            // OGN token can call `burnFrom(address(1))` later.abi
+
+            ogn.transfer(address(1), ognBalance);
+        }
+    }
+
+    /**
+     * @notice Returns the active status of the migration.
+     * @return True if migration has started and has not ended yet.
+     */
+    function isMigrationActive() public view returns (bool) {
+        return endTime > 0 && block.timestamp < endTime;
+    }
+
+    /**
+     * @notice Solvency Checks
+     *
+     * This ensures that the contract never transfers more than
+     * desired OGN amount in any case. This takes a balance diff
+     * of OGV and OGN and makes sure that the difference adds up.
+     * However, it doesn't revert if the difference is in favour
+     * of the contract (i.e. less OGN sent than expected).
+     */
+    modifier netBalanceCheck() {
+        uint256 ogvBalanceBefore = ogv.balanceOf(address(this));
+        uint256 ognBalanceBefore = ogn.balanceOf(address(this));
+
+        _;
+
+        uint256 netOgvAmountIn = ogv.balanceOf(address(this)) - ogvBalanceBefore;
+        uint256 netOgnAmountOut = ognBalanceBefore - ogn.balanceOf(address(this));
+
+        uint256 netExpectedOgnAmountOut = (netOgvAmountIn * CONVERSION_RATE / 1 ether);
+
+        if (netExpectedOgnAmountOut < netOgnAmountOut) {
+            // TODO: Do we need some sort of tolerance (may be 0.01%)?
+            revert BalanceMismatch(netOgvAmountIn, netExpectedOgnAmountOut, netOgnAmountOut);
+        }
+    }
+
+    /**
+     * @notice Migrates the specified amount of OGV to OGN
+     * @param ogvAmount Amount of OGV to migrate
+     * @return ognReceived OGN Received
+     */
+    function migrate(uint256 ogvAmount) external netBalanceCheck returns (uint256 ognReceived) {
+        return _migrate(ogvAmount, msg.sender);
+    }
+
+    /**
+     * @notice Migrates all of user's OGV to OGN
+     * @return ognReceived OGN Received
+     */
+    function migrateAll() external netBalanceCheck returns (uint256 ognReceived) {
+        return _migrate(ogv.balanceOf(msg.sender), msg.sender);
+    }
+
+    /**
+     * @notice Migrates caller's OGV to OGN and sends it to the `receiver`
+     * @return ognReceived OGN Received
+     */
+    function _migrate(uint256 ogvAmount, address receiver) internal returns (uint256 ognReceived) {
```

Comment:
Let's move this internal _migrate to the end of the two migrate methods. This way it's after the two publicly facing migrate methods.

Reply:
This is done as well